### ISSUE TITLE
[Bug]: The speed of compile mode does not perform well during parallel inference

### ISSUE THREAD

**Issue by ZJY0516** [vllm-project/vllm-omni#819]
### 🐛 Describe the bug
according to https://github.com/vllm-project/vllm-omni/pull/760
```bash
export DIFFUSION_ATTENTION_BACKEND=FLASH_ATTN # or TORCH_SDPA if not using FA
python examples/offline_inference/text_to_image/text_to_image.py --ulysses_degree 4 --height 1328 --width 1328 # this will pad hidden_states to the length divisble by ulysses_degree 

python examples/offline_inference/text_to_image/text_to_image.py --ulysses_degree 4 --height 1328 --width 1328 --enforce_eager # eager mode

```

- E2E inference

| attention  backend | torch.compile |time |
| --- | --- | ---|
| TORCH_SDPA |  TRUE |25.7s |
| FLASH_ATTN |  TRUE | 26.94s |
| TORCH_SDPA |  FALSE (eager mode)  |25.28s |
| FLASH_ATTN |  FALSE  (eager mode) | 21.70s |

> When torch.compile is enabled, FA is slower than SDPA.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

**Comment by wtomin** [comment]
I will take a look.

**Comment by wtomin** [comment]
**Hardware**: H800

Running the same inference experiments with the latest commit id `a3240fd4`:

attention backend | torch.compile | time
-- | -- | --
TORCH_SDPA | TRUE | 19.4s
FLASH_ATTN | TRUE |  15.9s
TORCH_SDPA | FALSE (eager mode) | 10.27s
FLASH_ATTN | FALSE (eager mode) |  7.84s


If changing the height and width to 1024x1024, which does not introduce token_padding and attention_mask like 1328x1328 does.

attention backend | torch.compile | time
-- | -- | --
TORCH_SDPA | TRUE | 3.62s
FLASH_ATTN | TRUE |  3.34s
TORCH_SDPA | FALSE (eager mode) | 5.0s
FLASH_ATTN | FALSE (eager mode) |  5.38s

This seems a lot more normal, although FLASH_ATTN is still a bit slower than TORCH_SDPA.

I think compilation may perform worse in padding&attention mask case. Let's keep tuned for UAA's progress and I will test it again afterwards. #1379 @dongbo910220

**Comment by dongbo910220** [comment]
> I think compilation may perform worse in padding&attention mask case. Let's keep tuned for UAA's progress and I will test it again afterwards. [#1379](https://github.com/vllm-project/vllm-omni/pull/1379) [@dongbo910220](https://github.com/dongbo910220)

Hopefully the elimination of padding and attention masks will resolve the compilation bottleneck.

**Comment by dongbo910220** [comment]
@wtomin 
 I use V100 * 4 to test and my #819 numbers aren’t 1:1 comparable because:

  - Switched to Z-Image Turbo on this machine due to VRAM constraints (different from the model used in #819).
  - This box is V100 (sm70), so FLASH_ATTN isn’t available here; we only benchmark TORCH_SDPA.
  - Z-Image has num_attention_heads=30, so --ulysses_degree 4 + ulysses_mode=strict can’t run (30 % 4 != 0). We can still cover the 4-GPU case with ulysses_mode=advanced_uaa (head padding 30→32). 

 4-GPU (ulysses=4, ring=1, TORCH_SDPA, advanced_uaa, steps=8):
| resolution | torch.compile | time |
| :--- | :--- | :--- |
| 1024x1024 | TRUE | 4.795s |
| 1024x1024 | FALSE (eager mode) | 5.127s |
| 960x960 | TRUE | 4.140s |
| 960x960 | FALSE (eager mode) | 4.470s |

2-GPU (ulysses=2, ring=1, TORCH_SDPA, 1024×1024, steps=8):
| mode | torch.compile | time |
| :--- | :--- | :--- |
| strict | TRUE (enforce_eager=False) | 5.740s |
| strict | FALSE (enforce_eager=True) | 6.344s |
| advanced_uaa | TRUE | 5.774s |
| advanced_uaa | FALSE (eager) | 6.418s |


On my machine, `torch.compile` consistently outperforms eager mode (enforce_eager=True) across these configurations. Note: I compile with static specialization (dynamic=False) because dynamic=True hit a TorchDynamo SymInt/Triton HOP issue on our stack.

**Comment by Gaohan123** [comment]
According to the discussion, the issue has been solved. I can reopen if needed.
